File: src/main/java/com/homechange/api/model/Offer.java
package com.homechange.api.model;

import javax.persistence.*;
import java.util.Date;

/**
 * Created by Nemanja on 5/16/17.
 *
 * Offer model
 */
@Entity
public class Offer {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Column(nullable = false)
	private Date startDate;
	@Column(nullable = false)
	private Date endDate;
	private String country;
	private String city;
	private String details;
	private OfferStatus status;
	@ManyToOne(fetch=FetchType.LAZY)
	@JoinColumn(name="user_id")
	private User user;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Date getStartDate() {
		return startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public String getCountry() {
		return country;
	}

	public void setCountry(String country) {
		this.country = country;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getDetails() {
		return details;
	}

	public void setDetails(String details) {
		this.details = details;
	}

	public OfferStatus getStatus() {
		return status;
	}

	public void setStatus(OfferStatus status) {
		this.status = status;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}
}

File: src/main/java/com/homechange/api/model/Offer.kt
package com.homechange.api.model

import javax.persistence.*
import java.util.Date

/**
 * Created by Nemanja on 5/16/17.

 * Offer model
 */
@Entity
data class Offer (

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null,

    @Column(nullable = false)
    var startDate: Date? = null,

    @Column(nullable = false)
    var endDate: Date? = null,

    var country: String? = null,
    var city: String? = null,
    var details: String? = null,
    var status: OfferStatus? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: User? = null,

    @OneToMany(mappedBy = "offer")
    var messages: List<Message>? = null,

    @OneToMany(mappedBy = "offer")
    var reviews: List<Review>? = null
)
